Clang build fails because of `CMAKE_CXX_STANDARD 11`

From PR https://github.com/loki-project/loki-network/pull/70:
> Building fails with clang with `CMAKE_CXX_STANDARD 11`: `lokinet/test/test_llarp_queue_manager.cpp:142:56: error: static_assert with no message is a C++17 extension [-Werror,-Wc++17-extensions]`

--- a/llarp/ev_kqueue.hpp
+++ b/llarp/ev_kqueue.hpp
@@ -215,7 +215,7 @@ namespace llarp
     llarp_tun_io* t;
     device* tunif;
     tun(llarp_tun_io* tio, llarp_ev_loop* l)
-        : ev_io(-1, new LossyWriteQueue_t("kqueue_tun_write", l))
+        : ev_io(-1, new LossyWriteQueue_t("kqueue_tun_write", l, l))
         , t(tio)
         , tunif(tuntap_init()){};
 

--- a/llarp/rpc.cpp
+++ b/llarp/rpc.cpp
@@ -102,7 +102,7 @@ namespace llarp
       }
 
       void
-      HandleError()
+      HandleError() override
       {
         handler({}, false);
       }

--- a/test/test_llarp_queue.cpp
+++ b/test/test_llarp_queue.cpp
@@ -82,7 +82,7 @@ popFrontTester(Args& args)
 
     args.startCond.notify_one();
 
-    args.runCond.wait(guard, [&args]() { return args.runSignal; });
+    args.runCond.wait(guard, [&args]() { return !!args.runSignal; });
   }
 
   for(;;)
@@ -104,7 +104,7 @@ pushBackTester(Args& args)
 
     args.startCond.notify_one();
 
-    args.runCond.wait(guard, [&args]() { return args.runSignal; });
+    args.runCond.wait(guard, [&args]() { return !!args.runSignal; });
   }
 
   for(size_t i = 0; i < args.iterations; ++i)
@@ -322,7 +322,7 @@ TEST(TestQueue, manyProducerManyConsumer)
   {
     threads[i] = std::thread(std::bind(&popFrontTester, std::ref(args)));
 
-    args.startCond.wait(lock, [&args, i]() { return args.count == (i + 1); });
+    args.startCond.wait(lock, [&]() { return args.count == (i + 1); });
   }
 
   for(size_t i = 0; i < numThreads; ++i)
@@ -330,8 +330,8 @@ TEST(TestQueue, manyProducerManyConsumer)
     threads[i + numThreads] =
         std::thread(std::bind(&pushBackTester, std::ref(args)));
 
-    args.startCond.wait(
-        lock, [&args, i]() { return args.count == (numThreads + i + 1); });
+    args.startCond.wait(lock,
+                        [&]() { return args.count == (numThreads + i + 1); });
   }
 
   args.runSignal++;

--- a/test/test_llarp_queue_manager.cpp
+++ b/test/test_llarp_queue_manager.cpp
@@ -139,7 +139,8 @@ struct QueueData
   std::uint32_t* m_states;  // Array of index states.
 };
 
-static_assert(sizeof(QueueData) == sizeof(QueueManager));
+static_assert(sizeof(QueueData) == sizeof(QueueManager),
+              "QueueData not updated");
 
 static constexpr uint32_t GENERATION_COUNT_SHIFT = 0x2;
 static constexpr uint32_t ELEMENT_STATE_MASK     = 0x3;

